package.json

```diff
@@ -53,11 +58,13 @@
     "gulp-uglify": "^1.4.0",
     "jasmine-reporters": "~1.0.0",
     "jest-cli": "^15.1.1",
+    "typescript": "^2.1.4",
     "vinyl-source-stream": "^1.1.0"
   },
   "scripts": {
     "build": "./build_releases.sh",
     "release": "./build_releases.sh && npm publish",
+    "pretest": "tsc integration/types/index.ts --noEmit",
```

Comment:
add a step to build `index.d.ts` why would you want tsc to be run on all machines and a requirement for all developers? I don't mind adding it to the CI. What's the benefit of running it as a pre-test?